Some reordering and renaming of functions

WMDE-Fisch opened the pull request:
Group them a bit according to topic
Rename cookie functions to address usage of localStorage

thiemowmde commented on DeepCat.js:
```diff
@@ -473,8 +473,18 @@
+		$( '.imeselector' ).css( 'z-index', 'auto' );
+	}
+
+	function disableImeAndSuggestions() {
```
Could be merged into a single `enableImeAndSuggestions( enabled )`, followed by `enabled === false ? -1 : 'auto'`.

WMDE-Fisch replied:
Good idea. Done. :-)

manicki commented on DeepCat.js:
```diff
@@ -473,14 +473,9 @@
-	function disableImeAndSuggestions() {
-		$( '.suggestions' ).css( 'z-index', -1 );
-		$( '.imeselector' ).css( 'z-index', -1 );
+	function setImeAndSuggestions( enable ) {
```
if we decide to do so (evil bool param!) I'd suggest to use verb "toggle" instead of "set" in the method name.

WMDE-Fisch replied:
done

thiemowmde replied:
Nice, `toggle…` was what I was looking for. Thanks.

manicki commented on DeepCat.js:
```diff
@@ -473,8 +473,13 @@
+
 	/**
-	 * Hash function for generating hint box cookie token.
+	 * Hash function for generating hint box setting token.
```
"hint box setting" sounds really mysterious if you don't know exactly what this is about
How about "hide hint (box)" instead of "hint box" all over the code?

WMDE-Fisch replied:
done

manicki commented on DeepCat.js:
```diff
@@ -495,45 +500,40 @@
 	}
 
-	function hasHintCookie() {
+	function hasHintSetting() {
```
and this could then be something like "shouldHideHint" or "shouldShowHint"

WMDE-Fisch replied:
shouldHideHint is a bit tricky due to the global shouldHideHints named it checkShouldHideHints now

Commit:
Some reordering and renaming of functions

Group them a bit according to topic
Rename cookie functions to address usage of localStorage

manicki commented:
if can't say "should hide hints" you could possibly go the other way round with "should show hints" but yeah, its ok.
Also "makeHintToken" is still mysterious to me but I stop ranting. I might submit some follow ups if I really can't stand the naming :)
